marketplace/deployer_util/config_helper.py

```diff
@@ -156,8 +171,268 @@ def properties_matching(self, definition):
     ]
 
 
+_SCHEMA_VERSION_1 = 'v1'
+_SCHEMA_VERSION_2 = 'v2'
+_SCHEMA_VERSIONS = [_SCHEMA_VERSION_1, _SCHEMA_VERSION_2]
+
+
+class SchemaXGoogleMarketplace:
+  """Accesses the top level x-google-markplace."""
+
+  def __init__(self, dictionary):
+    self._app_api_version = None
+    self._published_version = None
+    self._published_version_meta = None
+    self._images = None
+    self._cluster_constraints = None
+
+    self._schema_version = dictionary.get('schemaVersion', _SCHEMA_VERSION_1)
+    if self._schema_version not in _SCHEMA_VERSIONS:
+      raise InvalidSchema('Invalid schema version {}'.format(
+          self._schema_version))
+
+    if 'clusterConstraints' in dictionary:
+      self._cluster_constraints = SchemaClusterConstraints(
+          dictionary['clusterConstraints'])
+
+    if not self.is_v2():
+      return
+
+    if 'applicationApiVersion' not in dictionary:
```

Comment:
from this line to line 217: Are those checks to validate that this schema is a v2? If so, we could put all that in a private validation function to remove all that code from this constructor. what do you think?